### PR TITLE
feat(bench): bench embeddings

### PR DESCRIPTION
**Pull request by jon-chuang**
# Description

Fixes: https://github.com/jerryjliu/llama_index/issues/6804

Results:
image

Seq length honestly doesn't seem to matter, but this could be because BERT will be padded with empty tokens unlike autoregressive models...?

Note that we don't use token length, but string length.

## Type of Change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code

### CODE REVIEW

File: benchmarks/embeddings/bench_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        .load_data()[0]
+        .get_content()
+    )
+    content_length = len(content)
+
+    strings_per_loop = content_length / string_length
+    strings = []
+
+    for offset in range(0, int(num_strings / strings_per_loop) + 1):
+        ptr = offset
```
Comment (logan-markewich) [inline benchmarks/embeddings/bench_embeddings.py:23]: Curious, what does `ptr` stand for? pointer? Just trying to understand how this loop is working lol

File: benchmarks/embeddings/bench_embeddings.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            strings.append(content[ptr : ptr + string_length])
+            ptr += string_length
+            if len(strings) == num_strings:
+                break
+
+    return strings
+
+
+def get_max_seq_length(model):  # type: ignore
+    return model._langchain_embedding.client.max_seq_length  # type: ignore
```
Comment (logan-markewich) [inline benchmarks/embeddings/bench_embeddings.py:34]: does this only work with langchain embeddings I'm guessing?

File: benchmarks/embeddings/bench_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                        "hf/sentence-transformers/all-mpnet-base-v2",
+                        get_max_seq_length(embed_models[1]),
+                    ),
+                    (
+                        "hf/sentence-transformers/all-MiniLM-L6-v2",
+                        get_max_seq_length(embed_models[2]),
+                    ),
+                ]
+
+                skip_set = [0]  # skip openai
```
Comment (logan-markewich) [inline benchmarks/embeddings/bench_embeddings.py:92]: should the models/model name/seq length tuples maybe be passed into the function?

**Review by logan-markewich — APPROVED** [review]
just a few small points, but looks useful!

File: benchmarks/embeddings/bench_embeddings.py (hunk 1 above)
Comment (jon-chuang) [reply benchmarks/embeddings/bench_embeddings.py:23]: oh, it's just looping over the string. I guess we could implement it using range step size instead

File: benchmarks/embeddings/bench_embeddings.py (hunk 2 above)
Comment (jon-chuang) [reply benchmarks/embeddings/bench_embeddings.py:34]: only for sentence-transformer embedding lol

File: benchmarks/embeddings/bench_embeddings.py (hunk 3 above)
Comment (jon-chuang) [reply benchmarks/embeddings/bench_embeddings.py]: Yep, done.

**Pull request merged**